File: single-language-analysis/dataset-files/java/TestSmell013.java
// Original URL: https://github.com/bennn/NLP/blob/7d0c90feeaac03308791641dc6c864e88eba2ed0/TrieTest.java#L79-L99

public class TestSmell013 {
	@Test
	public void SimpleTestFrequencies(){
		simpleT.insertAll(Arrays.asList(toInsert));
		HashMap<String, Integer> wordMap = new HashMap<String, Integer>();
		for(int i = 0; i < toInsert.length; i++){
			if(wordMap.containsKey(toInsert[i])){
				int freq = wordMap.get(toInsert[i]);
				wordMap.put(toInsert[i], freq + 1);
			} else {
				wordMap.put(toInsert[i], 1);
			}
		}
		Collection<TrieNode> children = (Collection<TrieNode>)simpleT.getRoot().getChildren();
		if(children != null){
			Iterator<TrieNode> iterator = children.iterator();
			while(iterator.hasNext()) {
				TrieNode child = iterator.next();
				assertEquals(child.getFreq(), (int) wordMap.get(child.getKey()));
			}
		} 
	}
}

File: single-language-analysis/dataset-files/python/test_smell_013.py
import pytest

class TestTrie:
    def test_smell_013(self):
        simple_t.insert_all(list(to_insert))
        word_map = {}
        for i in range(len(to_insert)):
            if to_insert[i] in word_map:
                freq = word_map[to_insert[i]]
                word_map[to_insert[i]] = freq + 1
            else:
                word_map[to_insert[i]] = 1
        children = simple_t.get_root().get_children()
        if children is not None:
            iterator = iter(children)
            while True:
                try:
                    child = next(iterator)
                    assert child.get_freq() == word_map[child.get_key()]
                except StopIteration:
                    break